parser_library/src/workspaces/workspace.h

```diff
@@ -148,16 +148,24 @@ class workspace : public diagnosable_impl, public parse_lib_provider, public lsp
     void find_and_add_libs(
         std::string root, const std::string& path_pattern, processor_group& prc_grp, const library_local_options& opts);
 
+    bool is_config_file(const std::string& file_uri);
+    workspace_file_info parse_config_file();
+
     bool load_and_process_config();
     // Loads the pgm_conf.json and proc_grps.json from disk, adds them to file_manager_ and parses both jsons.
     // Returns false if there is any error.
-    bool load_config(config::proc_grps& proc_groups, config::pgm_conf& pgm_config, file_ptr& pgm_conf_file);
+    bool load_config(config::proc_grps& proc_groups,
+        config::pgm_conf& pgm_config,
+        file_ptr& proc_grps_file,
+        file_ptr& pgm_conf_file);
 
     bool is_wildcard(const std::string& str);
 
     // files, that depend on others (e.g. open code files that use macros)
     std::set<std::string> dependants_;
 
+    std::set<std::string> opened_files_;
```

Comment:
I guess we can't tell from the file_manager?

Reply:
As discussed, For the sake of history - `workspace` cannot rely on opened files tracked by `file manager` as there is only one `file manager` but multiple `workspaces`.